pdns/dnsname.hh

```diff
@@ -95,9 +96,8 @@ public:
   }
   DNSName(const DNSName& a) = default;
   DNSName(DNSName&& a) = default;
-  explicit DNSName(const char* p): DNSName(p, std::strlen(p)) {} //!< Constructs from a human formatted, escaped presentation
```

Comment:
Maybe keeping this constructor explicitly like it is may avoid some "strange" behaviour in case of "early" `\0`. for `\0` hiden before lenght in string I don't know if it's something expectable.

Reply:
hmm this may not help *because it fallbacked to the "general `DNSName(char*, size_t) ` but this constructor dealed with "early" `\0` multiple release shipped assuming first `\0` position gives `length` at least with `DNSName(char *)` construction.

Reply:
Maybe a test like: shall do the trick?

```cpp
BOOST_AUTO_TEST_CASE(test_early_null) {
  const char * test = "powerdns\0.com.";
  DNSName data(test);
  DNSName expected("powerdns");
  BOOST_CHECK_EQUAL(data, expected);
}
```

Reply:
this test added in `pdns/test-dnsname_cc.cc` passed both on this branch and on master ff43a0c2cbe44a7d0233d0c0446801efca1fc3ce just added the case in the middle of the file look like it runned had the number of test going from 237 to 238 with it.

Reply:
(had no employer pressure it can wait just had some idea to test it, figuring out if the early `\0` is already checked etc)